### ISSUE TITLE
Dead Lock if load Plugin's Classes during class transforming in LaunchClassLoader

### ISSUE THREAD

**Issue by anatawa12** [Teneted/Tenet#577]
**Describe the bug**
Dead Lock if loads Plugin's Classes during class transforming

**To Reproduce**
1. put [fixRTM 2.0.14](https://www.curseforge.com/minecraft/mc-mods/fixrtm/files/3062391), [Real Train Mod 2.4.21](https://www.curseforge.com/minecraft/mc-mods/realtrainmod/files/3061973), [NGTLib 2.4.18](https://www.curseforge.com/minecraft/mc-mods/ngtlib/files/3003745) into mods
2. put [discordSRV 1.20.0](https://github.com/DiscordSRV/DiscordSRV/releases/tag/v1.20.0) into plugins
3. launch server
4. now deadlock is caused.

**Expected behavior**
deadlock should not be caused.

**Screenshots**

**Info (please complete the following information):**
- Mod/Plugins Name: fixRTM, Real Train Mod, NGTLib, DiscordSRV
 - Mod/Plugins  Version: fixRTM 2.0.14, Real Train Mod 2.4.21, NGTLib 2.4.18, DiscordSRV 1.20.0

**Additional context**

[Here](https://gist.github.com/f0dbd9fca8bc76531ca07938bd82e8e9) is the thread dump.

**Why this deadlock is caused**
loads some classes with PluginClassLoader during loading some classes with LaunchClassLoader.

The DiscordSRV adds Filter of log4j and fixRTM calls log4j in IClassTransformer. 
DiscordSRV's Filter loads some classes. So it loads some classes in PluginClassLoader during loading some classes with LaunchClassLoader.

```
Thread1 Thread2
   |       |
   *----------- call LaunchClassLoader.loadClass
   |       |    this means get lock of LaunchClassLoader instance
   |       |
   |       *--- call PluginClassLoader.loadClass
   |       |    this means get lock of PluginClassLoader instance
   |       |
   |       *--- call LaunchClassLoader.loadClass as parent of PluginClassLoader
   |            this means wait for getting lock of LaunchClassLoader instance
   |       
   *----------- call PluginClassLoader.loadClass from Filter from some plugin.
                this means wait for getting lock of PluginClassLoader instance

 Now there is dead lock
```

**One way to avoid this problem**

~~ make PluginClassLoader.getClassLoadingLock returns same object as LaunchClassLoader.getClassLoadingLock. ~~ This fix is not enough.

**Comment by anatawa12** [comment]
@Mgazul Sorry, I found my fix is not enough so Could you re-open this issue?

**Comment by anatawa12** [comment]
@Mgazul The fix of this deadlock issue was reverted in #651. Will you fix this? if this issue is not going to fixed, could you add wontfix label to specify it?
